.github/workflows/tests.yml

```diff
@@ -0,0 +1,79 @@
+name: Test
+
+permissions:
+    contents: read
+
+on:
+  push:
+    branches:
+      - "**"
+  pull_request:
+    branches:
+      - main
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    if: >
+      (github.event_name == 'push' && github.ref != 'refs/heads/main') ||
+      (github.event_name == 'pull_request' && github.base_ref == 'main')
+    container:
+      image: python:3.12-slim
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Install dependencies
+        run: |
+          apt-get update && apt-get install -y shellcheck
+          pip install -r requirements.txt -r requirements-build.txt
+
+      - name: Show installed packages
+        run: pip freeze
+
+      - name: Run pylint
+        run: build_scripts/run_pylint.sh
+
+      - name: Run pymarkdownlint
+        run: build_scripts/run_markdownlint.sh
+
+      - name: Run shellcheck
+        run: build_scripts/run_shellcheck.sh
+
+      - name: Run flake8
+        run: flake8
+
+      - name: Run pytest
+        shell: bash
+        run: |
+          pytest --verbose -m "not integration and not smoke" --timer-top-n 10 -n auto --cov=decomposer --cov=maker_playground
```

Comment:
Add specific directories that pertain to this repo.
There might be a better way to do this by omission in a config file.